pkg/proxy/conntrack/conntrack.go

```diff
@@ -30,25 +30,6 @@ import (
 	utilnet "k8s.io/utils/net"
 )
 
-// Interface for dealing with conntrack
-type Interface interface {
```

Comment:
This is wrong; I think the bug here is that I failed to tag `fake.go` as linux-only

Reply:
interfaces *shouldn't* live in platform specific files, nothing in the fake file requires linux, right?

Reply:
conntrack itself is a linux-specific interface, and `pkg/proxy/conntrack` is only imported from files that are `go:build linux` (and if anything that _wasn't_ tagged `go:build linux` tried to import it, we would expect it to fail).

Reply:
tagging fake.go would make the existence of the package be OS-specific, which isn't great for tooling. *some* file in the package should be OS-independent... the interface and fake file seemed reasonable since they don't actually have any linux-specific code or dependencies

Reply:
> tagging fake.go would make the existence of the package be OS-specific, which isn't great for tooling. some file in the package should be OS-independent

If that's true we should enforce that; `pkg/proxy/iptables` and `pkg/proxy/nftables` are currently fully-linux-only as well. (As was `pkg/proxy/conntrack` was [before Monday](https://github.com/kubernetes/kubernetes/pull/122768).) `pkg/proxy/winkernel` is only "OS-independent" by virtue of `metrics.go` but I was planning to move that to `pkg/proxy/metrics` soon...

> the interface and fake file seemed reasonable since they don't actually have any linux-specific code or dependencies

Even the parts of `pkg/proxy/conntrack` that have no _visible_ linux-specific code or dependencies are still 100% linux-specific. (If we needed an OS-independent "connection tracking" interface in the future, it would not look like `conntrack.Interface`.)

Maybe just add an empty `doc.go` like in some other packages?

Reply:
ok, tried that